src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java

```diff
@@ -172,22 +171,21 @@ private AboutToSubmitCallbackResponse validateAndSetFromAboutToSubmitCallback(fi
 
         validateSignificantItem(aboutToSubmitCallbackResponse, callbackResponse);
         callbackService.validateCallbackErrorsAndWarnings(callbackResponse, ignoreWarning);
+        callbackResponse.updateCallbackStateBasedOnPriority();
+        aboutToSubmitCallbackResponse.setState(Optional.ofNullable(callbackResponse.getState()));
+        if (callbackResponse.getState() != null) {
+            caseDetails.setState(callbackResponse.getState());
+        }
```

Comment:
can you use `map` on the result of `callbackResponse.getState()` and inside the `map` you do both
`aboutToSubmitCallbackResponse.setState` and `caseDetails.setState`?